framework/tlvf/src/src/CmduMessageTx.cpp

```diff
@@ -78,4 +78,34 @@ bool CmduMessageTx::finalize()
     if (!addClass<tlvEndOfMessage>())
         return false;
     return msg.finalize();
-};
+}
+
+/**
+ * @brief Return how much element of given size can be allocated without create a MTU restriction
+ * overflow.
+ * 
+ * @param[in] size size of one element in bytes.
+ * @return size_t number of elements of 'size' that can be inserted.
+ */
+size_t CmduMessageTx::mtu_size_restriction_overflow_check(size_t size)
```

Comment:
Not the best method name.
Consider using something like `elements_in_message`

Reply:
change it to "element_in_message_mtu_limit" as a compromise. :)

Reply:
What if tomorrow we'll use another transport that is not limited to MTU size?
Don't see any reason to mention MTU in the method name...

Reply:
It is not the transport that limits us. It is IEEE1950.1.

Reply:
Whatever, there's still no reason to have "MTU" as part of the method name.
The users (developers that will call this method) shouldn't care about the underlying limitations, all they want is to know how many elements they can write into a message...

Reply:
I don't agree. We are bounded to 1905.1 restrictions. It is not something that will change,
and that is the limit the function uses to calculate how many elements we can add, so the developer actually does care about it. The reason that it is important is that the real limitation is the buffer size limitation, but for this specific message, we want to make sure we haven't crossed the MTU limitation.

Reply:
You're correct that we do not plan replacing 1905.1, but the MTU size is actually defined by the data-link layer (Ethernet in our case):
![image](https://user-images.githubusercontent.com/4477886/79995929-84833500-84c0-11ea-8e9d-15e31adb8b16.png)

I would accept your argument if it was an internal mechanism that would automatically split messages based on the actual MTU size, but you're forcing the user to make this calculation.
And since it's a user facing method, it should have a simple and user-friendly name.

Again, the user shouldn't care why there's a limitation on the number of elements (MTU or any other reason) in the message, but just that there is.

By the way, assuming that MTU is always 1500 bytes is also wrong, the hardware could support Jambo frames and then the MTU size is 9000 bytes...